app/src/test/java/org/simple/clinic/registration/confirmpin/RegistrationConfirmPinScreenControllerTest.kt

```diff
@@ -45,57 +50,47 @@ class RegistrationConfirmPinScreenControllerTest {
 
   @Test
   fun `when 4 digits are entered then the PIN should be submitted automatically`() {
-    whenever(userSession.ongoingRegistrationEntry()).thenReturn(OngoingRegistrationEntry(pin = "1234").toOptional())
+    whenever(userSession.ongoingRegistrationEntry()).thenReturn(ongoingEntry.toOptional())
 
     uiEvents.onNext(RegistrationConfirmPinTextChanged("1"))
     uiEvents.onNext(RegistrationConfirmPinTextChanged("12"))
     uiEvents.onNext(RegistrationConfirmPinTextChanged("123"))
     uiEvents.onNext(RegistrationConfirmPinTextChanged("1234"))
 
-    verify(userSession).saveOngoingRegistrationEntry(OngoingRegistrationEntry(
-        pin = "1234",
-        pinConfirmation = "1234",
-        createdAt = Instant.now(clock)))
+    verify(userSession).saveOngoingRegistrationEntry(ongoingEntry.withPinConfirmation("1234", clock))
   }
 
   @Test
   fun `when next is clicked with a matching PIN then ongoing entry should be updated and the next screen should be opened`() {
     val input = "1234"
 
-    val ongoingEntry = OngoingRegistrationEntry(pin = "1234")
     whenever(userSession.ongoingRegistrationEntry()).thenReturn(ongoingEntry.toOptional())
 
     uiEvents.onNext(RegistrationConfirmPinTextChanged(input))
 
     val inOrder = inOrder(userSession, screen)
-    inOrder.verify(userSession).saveOngoingRegistrationEntry(check {
-      assertThat(it.pinConfirmation).isEqualTo(input)
-      assertThat(it.createdAt).isNotNull()
-    })
+    inOrder.verify(userSession).saveOngoingRegistrationEntry(ongoingEntry.withPinConfirmation("1234", clock))
     inOrder.verify(screen).openFacilitySelectionScreen()
   }
 
   @Test
   fun `proceed button clicks should only be accepted if the confirmation pin matches with original pin`() {
-    val originalPin = "1234"
     val invalidConfirmationPin = "123"
     val validConfirmationPin = "1234"
 
-    val ongoingEntry = OngoingRegistrationEntry(pin = originalPin)
     whenever(userSession.ongoingRegistrationEntry()).thenReturn(ongoingEntry.toOptional())
 
     uiEvents.onNext(RegistrationConfirmPinTextChanged(invalidConfirmationPin))
     uiEvents.onNext(RegistrationConfirmPinDoneClicked())
 
     uiEvents.onNext(RegistrationConfirmPinTextChanged(validConfirmationPin))
 
-    verify(userSession, times(1)).saveOngoingRegistrationEntry(any())
```

Comment:
It has been removed because it is effectively the same as `verify(userSession, times(1)).saveOngoingRegistrationEntry()`. The extra `times(1)` is redundant information that serves no purpose.